server/channels/app/platform/web_hub.go

```diff
@@ -503,7 +522,13 @@ func (h *Hub) Start() {
 				for _, webConn := range connIndex.ForUser(userID) {
 					webConn.InvalidateCache()
 				}
-				connIndex.InvalidateCMCacheForUser(userID)
+				err := connIndex.InvalidateCMCacheForUser(userID)
+				if err != nil {
+					h.platform.Log().Error("Error while invalidating channel member cache", mlog.String("user_id", userID), mlog.Err(err))
+					for _, webConn := range connIndex.ForUser(userID) {
+						webConn.Close()
```

Comment:
This could be problematic as I think closing would call `(h *Hub) Unregister` before returning.

Reply:
You are right. Following the existing method of :

```
close(webConn.send)
connIndex.Remove(webConn)
```